lib/iterator.js

```diff
@@ -187,6 +187,23 @@ class Iterator {
     return new EnumerateIterator(this);
   }
 
+  // Consumes an iterator grouping values by keys
+  //   accessor <Function> gets value to group by
+  //     value <any> current iterator value
+  //     Returns: <any> value to group by
+  //   Returns: <Map> map with arrays of iterator values grouped
+  //       by keys returned by `accessor`
+  groupBy(accessor) {
+    const map = new Map();
+    for (const value of this) {
+      const key = accessor(value);
```

Comment:
Should we?
```js
      const key = accessor?.(value) ?? value;
```

Reply:
That will make this code incompatible with Node.js <14 which is not acceptable since we currently support >=8

Reply:
Yeah, I know, that is why it is not a ```suggestion.
The question is "Should we make it behave this way?".

Reply:
Oh, I thought about it initially but got to the conclusion that it must be explicit to be clear what is the grouping factor. Also, I don't think it is very useful to group by iterator value anyway.

Reply:
The main reason is to be consistent with the accessor in `findCompare`, `max` and `min`.

Reply:
Well, this is a different use-case which IMO doesn't benefit from such default case compared to those other functions.